mypy/semanal.py

```diff
@@ -4786,7 +4786,32 @@ def add_imported_symbol(self,
                             module_hidden: bool) -> None:
         """Add an alias to an existing symbol through import."""
         assert not module_hidden or not module_public
-        symbol = SymbolTableNode(node.kind, node.node,
+
+        symbol_node: Optional[SymbolNode] = node.node
+        # I promise this type checks; I'm just making mypyc issues go away.
+        # mypyc is absolutely convinced that `symbol_node` narrows to a Var in the following,
+        # when it can also be a FuncBase.
+        symbol_node_any: Any = cast(Any, symbol_node)
+        if self.is_class_scope() and isinstance(symbol_node, (FuncBase, Var)):
```

Comment:
I had a similar problem, see https://github.com/mypyc/mypyc/issues/892 and https://github.com/python/mypy/pull/11150

Reply:
Thanks, yeah, that looks very similar